src/Compilers/CSharp/Portable/Symbols/Attributes/AttributeData.cs

```diff
@@ -41,10 +41,6 @@ internal abstract partial class CSharpAttributeData : AttributeData
         /// </summary>
         public new abstract SyntaxReference? ApplicationSyntaxReference { get; }
 
-        // Overridden to be able to apply MemberNotNull to the new members
-        [MemberNotNullWhen(true, nameof(AttributeClass), nameof(AttributeConstructor))]
```

Comment:
> [MemberNotNullWhen(true, nameof(AttributeClass), nameof(AttributeConstructor))] [](http://example.com/codeflow?start=7&length=80)

I think this attribute was referring to different members by comparison to the attribute in base. Consider keeping the the attribute.

Reply:
I don't see a need to keep an attribute given that it doesn't appear to benefit any consumers of this type.